### PR TITLE
Glossary

### PR DESCRIPTION
**Pull request by omichel**
Addresses https://github.com/omichel/webots/issues/4004

### CODE REVIEW

**Comment by omichel** [comment]
:+1: (feel free to add more definitions or correct or extend the existing ones)

**Comment by omichel** [comment]
:+1:

**Comment by stefaniapedrazzi** [comment]
I will also add the definition of ``ancestor``, ``child`` and ``parent`` node.

**Comment by stefaniapedrazzi** [comment]
A also changed *descendant* to *descendant node* given that the description explicitly refers to a node and not to the generic descendant terms (like for *kinemwtic solid*).

**Comment by stefaniapedrazzi** [comment]
:+1:

**Comment by omichel** [comment]
I don't fully agree on adding the *node* suffix to *descendant*, *ancestor*, *child* and *parent*. Because unlike *kinematic* or *dynamic*, these are nouns and not adjectives and it is lighter to say "the descendant of..." or the "parent of..", etc. than "the parent node of..." especially because these nouns apply on to nodes in Webots (and not to field or other concepts). For example our methods usually don't use the *node* suffix and that's right, like getParent() instead of getParentNode(), etc.

**Comment by stefaniapedrazzi** [comment]
Ok, but then we should specify in the description that the description we provide is only applicable in case we refer to a node. I think it is wrong to given a generic definition of ancestor, descendant, parent and child and assume we are speaking about nodes. It could also apply to fields...

**Comment by stefaniapedrazzi** [comment]
I mean the definition of "descendant" in the glossary it is not correct for me. Because we provide the specific definition of "descendant node" but it is wrong in case of "descendant field".

**Comment by omichel** [comment]
Do we really need the concept of "descendant field"? That seems a bit overkill to me...

File: reference/glossary.md [hunk 1]
```diff
@@ -4,13 +4,17 @@ This glossary defines the terminology used to describe the various concepts rela
 
 **Actuator**: An *actuator* is a *node* representing a real robotics actuator such as a motor, a LED, a radio emitter, etc. *Actuators* may receive commands from a *controller*. Their behavior will affect the outcome of a simulation. Some *actuators* are also *sensors* if they make sensory measurement, such as a motor than can provide position or force feedback.
 
+**Ancestor node**: An *ancestor node* is a *node* that hierarchically includes another *node* in the *scene tree*.
+
```
Comment (omichel) [inline reference/glossary.md]: I would rephrase it relatively to the descendant node, e.g.:

**Ancestor**: The *ancestor* of a *descendant* is a *node* which hierarchically contains the *descendant* in the *scene tree*.

Comment (stefaniapedrazzi) [inline reference/glossary.md]: It sound wrong to use ancestor and descendant without the word node. We never do it in the documentation.

If you want to define the ancestor node based on the descendant node definition is ok for me.
But this definition doesn't seems very consistent with descendant node description.
Then I would prefer something like:
*The converse notion of a descendant node.*

Comment (omichel) [inline reference/glossary.md]: Yes, I agree that we should also rephrase the definition of the descendant node, so that it is consistent.

**Comment by stefaniapedrazzi** [comment]
We don't need the concept of descendant field, but the word descendant in a tree could apply both to fields and nodes. The same for ancestor, child and parent and we should be consistent without the definitions.
In the scene tree the 'translation' field is a child of a 'Transform' node.
So for me a 'child' (or 'descendant') cannot be assumes to be a node.

**Comment by omichel** [comment]
I would rather say that 'translation' is a field of the 'Transform' node, not a child...

**Comment by omichel** [comment]
Let me speak with David to get his point of view and I will call you on Skype to finalize this discussion.

**Comment by stefaniapedrazzi** [comment]
> I would rather say that 'translation' is a field of the 'Transform' node, not a child...

If you look the the hierarchy as a pure tree, then the field item is a child.
You would say that it is a field, but what will the user say at first sight?

**Comment by omichel** [comment]
@stefaniapedrazzi: may I let you review my changes and let me know if you agree?

**Comment by omichel** [comment]
:+1:

**Comment by stefaniapedrazzi** [comment]
For me it is fine, even if I miss the reference to the 'field' intermediate structure when describing the *child* and *parent* terms.

:+1:

**Comment by omichel** [comment]
I added a definition for *field*.

:+1:

**Comment by stefaniapedrazzi** [comment]
:+1:

**Comment by fabienrohrer** [comment]
👍

File: reference/glossary.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Glossary
+
+This glossary defines the terminology used to describe the various concepts related to a Webots simulation.
+
+**Actuator**: An *actuator* is a *node* representing a real robotics actuator such as a motor, a LED, a radio emitter, etc. *Actuators* may receive commands from a *controller*. Their behavior will affect the outcome of a simulation. Some *actuators* are also *sensors* if they make sensory measurement, such as a motor than can provide position or force feedback.
```
Comment (DavidMansolino) [inline reference/glossary.md]: > Some *actuators* are also *sensors* if they make sensory measurement, such as a motor than can provide position or force feedback.

I have a doubt about this sentence, because motor does not provide position measurement anymore, and for me the force-feedback is kind of an exception.

Comment (omichel) [inline reference/glossary.md]: You are right. I will remove the position feedback from this sentence. But although it is an exception, I believe we should keep this comment about force feedback.
